index.js

```diff
@@ -17,46 +16,60 @@ var tree = quadtree(0, 0, gameConfig.width, gameConfig.height);
 var players = [];
 var men = [];
 var sockets = {};
-
-function addMan (manToAdd, socketId) {
-	manToAdd.id = socketId;
-	manToAdd.x = utils.getRandomNum();
-	manToAdd.y = utils.getRandomNum();
-	men.push(manToAdd);
-	tree.put(manToAdd);
-}
-
-function removeMan (manToRemove) {
-	var removeIndex = men.indexOf(manToRemove);
-	men.splice(removeIndex, 1);
-	tree.remove(manToRemove);
-}
+var removedPlayers = 0; // once it reaches 100 garbage COLLECTION!
+
+// function addMan (manToAdd, socketId) {
+// 	manToAdd.id = socketId;
+// 	manToAdd.x = utils.getRandomNum();
+// 	manToAdd.y = utils.getRandomNum();
+// 	men.push(manToAdd);
+// 	tree.put(manToAdd);
+// }
+
+// function removeMan (manToRemove) {
+// 	var removeIndex = men.indexOf(manToRemove);
+// 	men.splice(removeIndex, 1);
+// 	tree.remove(manToRemove);
+// }
 
 function addPlayer(playerData, socketId) {
-	playerData.id = socketId
+	playerData.id = socketId;
 	players.push(playerData);
 }
 
-function moveLoop () {
-	for (var i = 0; i < men.length; i++) {
-		trackMan(men[i]);
-	}
+function removePlayer (socket) {
```

Comment:
the actual player removal function. removes them thru deletion from objects. 

removes them from the players array by setting them to null. 
when the number reaches 100, it cleans them in one go with the garbage collection function that filters out elements === null.